core/src/main/scala/kafka/server/NodeToControllerChannelManager.scala

```diff
@@ -52,37 +51,6 @@ trait ControllerNodeProvider {
   def getControllerInfo(): ControllerInformation
 }
 
-class MetadataCacheControllerNodeProvider(
-  val metadataCache: ZkMetadataCache,
-  val config: KafkaConfig,
-  val quorumControllerNodeProvider: () => Option[ControllerInformation]
-) extends ControllerNodeProvider {
-
-  private val zkControllerListenerName = config.interBrokerListenerName
-  private val zkControllerSecurityProtocol = config.interBrokerSecurityProtocol
-  private val zkControllerSaslMechanism = config.saslMechanismInterBrokerProtocol
-
-  val emptyZkControllerInfo =  ControllerInformation(
-    None,
-    zkControllerListenerName,
-    zkControllerSecurityProtocol,
-    zkControllerSaslMechanism,
-    isZkController = true)
```

Comment:
Could you please remove `isZkController` and `isNetworkClientForZkController` from this class file?